src/main/java/ch/njol/skript/patterns/OptionalPatternElement.java

```diff
@@ -52,6 +61,8 @@ public PatternElement getPatternElement() {
 
 	@Override
 	public String toString() {
+		if (negated)
+			return "";
```

Comment:
I'm unsure about this. toString for patterns is supposed to return their *raw* form.
It seems a little *too* hacky to me, and I do wonder if we are just better off allowing two strings for property registration (`property` and `negatedProperty`). For your example, this would look something like:
```java
String property = " [direct] line of sight"
register(CondHasLineOfSight.class, PropertyType.HAVE, "a" + property, "no" + property, "entities/locations");
```
I think a solution like this would be preferred.

Reply:
Cannot be done due to the ending of the register method already being a VarArg of strings. Would not allow for past compatibility, and with the addition of a new pattern it can be used literally anywhere where a Skript pattern is accepted.

Reply:
The register method is *not* a vararg of strings, and a new overloaded method can be added with both parameters (no compatibility issues). Instead of obtaining the negated string from the pattern, the user would simply provide it. I think this existing implementation is a bit of a misuse of the pattern system. Notably, we are hardcoding in something for a utility-type class, but also the fact that it goes against the conventions for the toString method, which is that it should return a string representing the original pattern provided.

I really think we should explore ways of achieving this *without* having to make changes to the pattern system.